src/Controller.php

```diff
@@ -34,7 +34,13 @@ private function getProvider($provider, $redirectUri = null)
     {
         $config = $this->getConfig($provider) + compact('redirectUri');
 
-        $class  = 'League\\OAuth2\\Client\\Provider\\' . ucfirst($provider);
+        if (isset($_ENV[strtoupper($provider) . '_NAMESPACE'])) {
+            $namespace = $_ENV[strtoupper($provider) . '_NAMESPACE'];
```

Comment:
I was trying to avoid that, but since they're all listed in composer anyways, I guess it doesn't matter.